ProductWarehouse.Persistence.EF/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ProductWarehouse.Application.Interfaces;
+using ProductWarehouse.Domain.Entities;
+using ProductWarehouse.Persistence.Abstractions;
+using ProductWarehouse.Persistence.Abstractions.Exceptions;
+using Serilog;
+
+namespace ProductWarehouse.Persistence.EF.Repositories;
+
+public class ProductRepository : Repository<Product>, IProductRepository
+{
+	private readonly ApplicationDbContext _dbContext;
+	private readonly ILogger _logger;
+
+	public ProductRepository(ApplicationDbContext dbContext, ILogger logger) : base(dbContext, logger)
+	{
+		_dbContext = dbContext;
+		_logger = logger;
+	}
+
+	public async Task<List<Product>> GetProductsAsync()
+	{
+		try
+		{
+			var products = await _dbContext.Products
+				.Where(x => !x.IsDeleted)
+				.Include(p => p.Brand)
+				.Include(p => p.ProductGroups).ThenInclude(pg => pg.Group)
+				.Include(p => p.ProductSizes).ThenInclude(pg => pg.Size)
+				.ToListAsync();
+
+			return products;
+		}
+		catch (InvalidOperationException ex)
+		{
+			_logger.Warning("No products found.", ex);
+			throw new NotFoundException("No products found.", ex);
```

Comment:
The messages for entities not found must be the same so as not to confuse the client who consumes us